experimental/modular_oak_functions/src/main.rs

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+pub mod demux;
+pub mod io;
+pub mod log;
+pub mod lookup;
+pub mod policy;
+pub mod session;
+pub mod wasmi;
+
+use maplit::btreemap;
+use std::{collections::BTreeMap, sync::Arc};
+
+/// Manages the shared state for a service.
+pub trait Service {
+    /// Creates a new service proxy
+    fn create_proxy(&self) -> Box<dyn ServiceProxy>;
+
+    /// Configures the servcice with the given data.
+    ///
+    /// Each service is responsible for interpreting the bytes approriately.
+    ///
+    /// Most services should only react to receiving configuration data once, except for the lookup
+    /// service which will periodically receive lookup data refreshes through this interface.
+    fn configure(&self, data: &[u8]) -> anyhow::Result<()>;
```

Comment:
When is a service expected to be configured? Once, at startup time? Repeatedly? Where would this method be invoked?